extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -1063,21 +1183,32 @@ static class CodecTransfer<T> implements TransferObject<Binary[]> {
         private ObjectChunk<T, Values> chunk;
         private final Binary[] buffer;
         private final ColumnSource<T> columnSource;
-
+        private boolean hasMoreDataToBuffer;
 
         CodecTransfer(ColumnSource<T> columnSource, ObjectCodec<? super T> codec, int targetSize) {
             this.columnSource = columnSource;
             this.buffer = new Binary[targetSize];
             context = this.columnSource.makeGetContext(targetSize);
             this.codec = codec;
+            hasMoreDataToBuffer = false;
         }
 
         @Override
-        public void propagateChunkData() {
+        public boolean hasMoreDataToBuffer() {
+            return hasMoreDataToBuffer;
+        }
+
+        @Override
+        public int bufferFetchedData() {
+            if (!hasMoreDataToBuffer) {
```

Comment:
The binaries generated from the codec also are variable length and would benefit from the page size tests (similar to String).